kubernetes/ingress-nginx
# Chart: Add `TargetGroupBinding`.

## What this PR does / why we need it:


Add a TargetGroupBinding object to the ingress-nginx helm chart template.
When creating an ingress-nginx controller, it is necessary to use an already existing AWS load balancer.
To use this, the controller's service must be a nodePort type, and aws-load-balancer-controller must already be installed in the cluster.

## Types of changes

- [ ] CVE Report (Scanner found CVE and adding report)

## Which issue/s this PR fixes


## How Has This Been Tested?



1. Create aws nlb and create a `tcp:30004` & instance type target group that will be the forwarding target for 80 & 443 listeners.
2. Run helm install with the controller.service property below.
```yaml
controller:
  service:
    enabled: true
    enableHttp: true
    enableHttps: true
    type: NodePort
    targetPorts:
      http: http
      https: http
    nodePorts:
      http: 30004
      https: null
   
    awsTargetGroupBinding:
      enabled: true
      arnMapping:
        http: <arn-to-targetGroup>
        https: <arn-to-targetGroup>
```
3. Check the AWS target group - check whether the nodes are attached well.
4. Please also note that the security group of the eks node group must allow the security group of nlb!
## Checklist:


- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [ ] I have added unit and/or e2e tests to cover my changes.
